src/api/meal_plan.py

```diff
@@ -1,89 +1,227 @@
-            }
+            result = connection.execute(
+                text("""
+                    INSERT INTO meal_plans (author_id, name, description)
+                    VALUES (:account_id, :name, :description)
+                    RETURNING id
+                """),
+                {
+                    "account_id": plan.account_id,
+                    "name": plan.name,
+                    "description": plan.description
+                }
+            )
+
+            meal_plan_id = result.scalar()
+            return {"meal_plan_id": meal_plan_id}
+    
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=f"{e}")
+
+@router.delete("/{meal_plan_id}")
+def delete_meal_plan(meal_plan_id: int):
+    try:
+        with db.engine.begin() as connection:
+            validate = connection.execute(
+                text("""SELECT 1 FROM meal_plans WHERE id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            ).one_or_none()
+
+            if validate is None:
+                print(f"Meal Plan with ID {meal_plan_id} not found.")
+                raise Exception("Meal Plan not found.")
+
+            connection.execute(
+                text("""DELETE FROM plans_recipes WHERE meal_plan_id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            )
+            connection.execute(
+                text("""DELETE FROM meal_plans WHERE id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            )
+
+            return Response(status_code=200)
+    
+    except Exception as e:
+        raise HTTPException(
+            status_code=404, detail=f"An error occurred while deleting the meal plan: {e}"
         )
-    return {"message": "Meal logged successfully"}
 
-@router.get("/meal-plans/{meal_plan_id}/calories")
-def get_daily_calories(meal_plan_id: int, date: str):
+@router.post("/{meal_plan_id}/recipes/{recipe_id}")
+def add_recipes(meal_plan_id: int,recipe_id: int):
+    try: 
+        with db.engine.begin() as connection:
+
+            validate_meal_plan = connection.execute(text("""SELECT 1 FROM meal_plans WHERE id = :id"""), {"id": meal_plan_id}).one_or_none()
+            if validate_meal_plan is None:
+                raise Exception("Meal plan not found.")
+            
+            validate_recipe = connection.execute(text("""SELECT 1 FROM recipes WHERE id = :id"""), {"id": recipe_id}).one_or_none()
+
+            if validate_recipe is None:
+                raise Exception("Recipe not found.")
+            
+            connection.execute(text("""INSERT INTO plans_recipes (meal_plan_id, recipe_id) VALUES (:meal_plan_id, :recipe_id)"""), {"meal_plan_id": meal_plan_id, "recipe_id": recipe_id})
+
+            return Response(status_code=200)
+        
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=f"Error adding recipe: {e}")
+
+@router.delete("/{meal_plan_id}/recipes/{recipe_id}")
```

Comment:
Same as adding a recipe, I think recipe id should be included in the request body rather than URL. Would allow for multiple deletions/insertions in a single request as well